hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -181,6 +181,7 @@ class DefaultSource extends RelationProvider
       HoodieSparkSqlWriter.bootstrap(sqlContext, mode, optParams, dfWithoutMetaCols)
     } else {
       HoodieSparkSqlWriter.write(sqlContext, mode, optParams, dfWithoutMetaCols)
+      HoodieSparkSqlWriter.cleanup()
```

Comment:
i think a better place to shutdown metrics is `commitAndPerformPostOperations` in `HoodieSparkSqlWriter`

Reply:
The reason for doing it here is that, `HoodieSparkSqlWriter` is also used by `HoodieStreamingSink`. We don't want to be closing the reporter for the spark streaming job.